build/rules.mak

```diff
@@ -77,13 +77,23 @@ $(OUTPUT_DIR)/setup.py: $(TEMPLATE_DIR)/setup.py.mako
 	$(call trace_to_console, "\ \ \ \ Generating",$@)
 	$(_hide_cmds)$(call log_command,$(call GENERATE_SCRIPT, $<, $(dir $@), $(METADATA_DIR)))
 
-sdist: $(OUTPUT_DIR)/setup.py $(OUTPUT_DIR)/README.rst $(MODULE_FILES) $(LOG_DIR)/tests_passed
+sdist: $(SDIST_BUILT)
+
+$(SDIST_BUILT): $(OUTPUT_DIR)/setup.py $(OUTPUT_DIR)/README.rst $(MODULE_FILES) $(UNIT_TESTS_PASSED)
 	$(call trace_to_console, "Creating sdist",$(OUTPUT_DIR)/dist)
+	$(_hide_cmds)$(call log_command,touch $@)
+	$(_hide_cmds)$(call log_command,rm $@)
 	$(_hide_cmds)$(call log_command,cd $(OUTPUT_DIR) && python3 setup.py sdist $(LOG_OUTPUT) $(LOG_DIR)/sdist.log)
+	$(_hide_cmds)$(call log_command,touch $@)
+
+wheel: $(WHEEL_BUILT)
 
-wheel: $(OUTPUT_DIR)/setup.py $(OUTPUT_DIR)/README.rst $(MODULE_FILES) $(LOG_DIR)/tests_passed
+$(WHEEL_BUILT): $(OUTPUT_DIR)/setup.py $(OUTPUT_DIR)/README.rst $(MODULE_FILES) $(UNIT_TESTS_PASSED)
 	$(call trace_to_console, "Creating wheel",$(OUTPUT_DIR)/dist)
+	$(_hide_cmds)$(call log_command,touch $@)
```

Comment:
Lot's of duplication for this touch - rm - do something - touch pattern.
We should put in a function.

Reply:
Moved into make_with_tracking_file. I am open to a better name.